Content.Server/GameTicking/Rules/SuspicionRuleSystem.cs

```diff
@@ -0,0 +1,413 @@
+
+        SoundSystem.Play(filter, _addedSound.GetSound(), AudioParams.Default);
+
+        _doorSystem.AccessType = DoorSystem.AccessTypes.AllowAllNoExternal;
+
+        _checkTimerCancel = new CancellationTokenSource();
+        Timer.SpawnRepeating(DeadCheckDelay, CheckWinConditions, _checkTimerCancel.Token);
+    }
+
+    public override void Removed()
+    {
+        _doorSystem.AccessType = DoorSystem.AccessTypes.Id;
+        EndTime = null;
+        _traitors.Clear();
+
+        _playerManager.PlayerStatusChanged -= PlayerManagerOnPlayerStatusChanged;
+
+        _checkTimerCancel.Cancel();
+    }
+
+    private void CheckWinConditions()
+    {
+        if (!Enabled || !_cfg.GetCVar(CCVars.GameLobbyEnableWin))
+            return;
+
+        var traitorsAlive = 0;
+        var innocentsAlive = 0;
+
+        foreach (var playerSession in _playerManager.ServerSessions)
+        {
+            if (playerSession.AttachedEntity is not {Valid: true} playerEntity
+                || !_entities.TryGetComponent(playerEntity, out MobStateComponent? mobState)
+                || !_entities.HasComponent<SuspicionRoleComponent>(playerEntity))
+            {
+                continue;
+            }
+
+            if (!mobState.IsAlive())
+            {
+                continue;
+            }
+
+            var mind = playerSession.ContentData()?.Mind;
+
+            if (mind != null && mind.HasRole<SuspicionTraitorRole>())
+                traitorsAlive++;
+            else
+                innocentsAlive++;
+        }
+
+        if (innocentsAlive + traitorsAlive == 0)
+        {
+            _chatManager.DispatchServerAnnouncement(Loc.GetString("rule-suspicion-check-winner-stalemate"));
+            EndRound(Victory.Stalemate);
+        }
+
+        else if (traitorsAlive == 0)
+        {
+            _chatManager.DispatchServerAnnouncement(Loc.GetString("rule-suspicion-check-winner-station-win"));
+            EndRound(Victory.Innocents);
+        }
+        else if (innocentsAlive == 0)
+        {
+            _chatManager.DispatchServerAnnouncement(Loc.GetString("rule-suspicion-check-winner-traitor-win"));
+            EndRound(Victory.Traitors);
+        }
+        else if (_timing.CurTime > _endTime)
+        {
+            _chatManager.DispatchServerAnnouncement(Loc.GetString("rule-suspicion-traitor-time-has-run-out"));
+            EndRound(Victory.Innocents);
+        }
+    }
+
+    private enum Victory
+    {
+        Stalemate,
+        Innocents,
+        Traitors
+    }
+
+    private void EndRound(Victory victory)
+    {
+        string text;
+
+        switch (victory)
+        {
+            case Victory.Innocents:
+                text = Loc.GetString("rule-suspicion-end-round-innocents-victory");
+                break;
+            case Victory.Traitors:
+                text = Loc.GetString("rule-suspicion-end-round-trators-victory");
```

Comment:
misspelled?

Reply:
Seems this has always been misspelled, fixed now.